src-plugins/vtkDataMesh/readers/vtkDataMeshReader.cpp

```diff
@@ -86,140 +85,110 @@ bool vtkDataMeshReader::read(const QString& path) {
-                return false;
-            }
+            qDebug() << metaObject()->className() << ": " << vtkErrorCode::GetStringFromErrorCode(error);
+            return false;
         }
-        else
+        catch (...)
         {
+            qDebug() << metaObject()->className() << ": error reading " << path;;
             return false;
         }
     }
-
-    setProgress(100);
-    return true;
+    else
+    {
+        return false;
+    }
 }
 
-QStringList vtkDataMeshReader::metaDataKeysToCopy()
+bool vtkDataMeshReader::extractMetaData(QString path, vtkMetaDataSet* dataSet)
 {
-    QStringList keys;
-
-    keys << medMetaDataKeys::PatientID.key()
-         << medMetaDataKeys::PatientName.key()
-         << medMetaDataKeys::Age.key()
-         << medMetaDataKeys::BirthDate.key()
-         << medMetaDataKeys::Gender.key()
-         << medMetaDataKeys::Description.key()
-         << medMetaDataKeys::StudyID.key()
-         << medMetaDataKeys::StudyDicomID.key()
-         << medMetaDataKeys::StudyDescription.key()
-         << medMetaDataKeys::Institution.key()
-         << medMetaDataKeys::Referee.key()
-         << medMetaDataKeys::StudyDate.key()
-         << medMetaDataKeys::StudyTime.key()
-         << medMetaDataKeys::Performer.key()
-         << medMetaDataKeys::Report.key()
-         << medMetaDataKeys::Protocol.key()
-         << medMetaDataKeys::Origin.key()
-         << medMetaDataKeys::AcquisitionDate.key()
-         << medMetaDataKeys::AcquisitionTime.key()
-         << medMetaDataKeys::Modality.key()
-         << medMetaDataKeys::Orientation.key();
-
-    return keys;
+    if (extractMetaDataFromFieldData(dataSet) || extractMetaDataFromHeader(path, dataSet) || extractCartoMetaData(dataSet))
+    {
+        return true;
+    }
+    else
+    {
+        return false;
+    }
 }
 
-void vtkDataMeshReader::parseHeaderVtk(QString header, medAbstractData* medData)
+bool vtkDataMeshReader::extractMetaDataFromFieldData(vtkMetaDataSet* dataSet)
 {
-    QStringList list = header.split("\t");
+    bool foundMetaData = false;
+    vtkFieldData* fieldData = dataSet->GetDataSet()->GetFieldData();
 
-    if (list.count() > 1) // Regular VTK from MUSIC
+    for (int i = 0; i < fieldData->GetNumberOfArrays(); i++)
     {
-        QStringList keyList = metaDataKeysToCopy();
-        foreach(QString key, keyList)
+        QString arrayName = fieldData->GetArrayName(i);
+        QString prefix = "medMetaData::";
+
+        if (arrayName.startsWith(prefix))
         {
-            if (list.contains(key))
+            foundMetaData = true;
+            vtkStringArray* array = static_cast<vtkStringArray*>(fieldData->GetAbstractArray(i));
+            QString metaDataKey = arrayName.remove(0, prefix.length());
+
+            for (int j = 0; j < array->GetSize(); j++)
             {
-                medData->setMetaData(key, list.at(list.indexOf(key)+1));
+                data()->addMetaData(metaDataKey, QString(array->GetValue(j)));
```

Comment:
```addMetadata ``` because the first time it's called, there is not metadata set yet (so it equals a ```setMetaData```) and the following times we don't want to overwrite the previous values.